python/ray/_private/node.py

```diff
@@ -157,6 +157,13 @@ def __init__(
         if not self.head:
             self.validate_ip_port(self.address)
             self._init_gcs_client()
+            self._cluster_metadata = self._read_cluster_info_from_kv()
+
+        # Fetch the head OS platform
+        if head:
+            self._head_os = sys.platform
+        else:
+            self._head_os = self._cluster_metadata["os"]
```

Comment:
`_cluster_metadata` and `_head_os` are added to node class fields and ensured to exist on both head and worker. 
 `_cluster_metadata` field in head node are set through return value of `put_cluster_metadata`:

https://github.com/ray-project/ray/blob/9ebd1b25721d215985bfd6bf32e833b1215b21fc/python/ray/_private/node.py#L1310-L1312